src/function.ts

```diff
@@ -760,13 +762,18 @@ export class CallbackLambdaCode extends aws_lambda.Code {
                   // @ts-ignore - private - adds the tag manager, which we don't need
                   cfnProperties,
                   ...rest
-                } = transformTable(o as CfnResource);
+                } = transformBus(transformTable(o as CfnResource));
                 return transformTaggableResource(rest);
               } else if (Token.isUnresolved(o)) {
-                const token = (<any>o).toString();
-                // add to tokens to be turned into env variables.
-                tokens = [...tokens, token];
-                return token;
+                const reversed = Tokenization.reverse(o)!;
+                if(Reference.isReference(reversed)) {
+                  tokens.push(reversed.toString());
+                  return reversed.toString();
+                } else if("value" in reversed) {
+                  return (reversed as unknown as {value: any}).value;
+                }
+                // TODO: fail at runtime and warn at compiler time when a token cannot be serialized
```

Comment:
What cases are there?

Reply:
good question, part of the problem is that I don't know. `LazyAny` is one.

Reply:
Probably more... but what I found: https://github.com/functionless/functionless/pull/243#issuecomment-1157249992